packages/hooks/src/utils/useBaseQuery.ts

```diff
@@ -0,0 +1,48 @@
+import { useContext, useEffect, useReducer, useRef } from 'react';
+import { getApolloContext, OperationVariables } from '@apollo/react-common';
+import { DocumentNode } from 'graphql';
+
+import { QueryHookOptions, QueryOptions } from '../types';
+import { QueryData } from '../data/QueryData';
+import { useDeepMemo } from './useDeepMemo';
+
+export function useBaseQuery<TData = any, TVariables = OperationVariables>(
+  query: DocumentNode,
+  options?: QueryHookOptions<TData, TVariables>,
+  lazy?: boolean
+) {
+  const context = useContext(getApolloContext());
+  const [tick, forceUpdate] = useReducer(x => x + 1, 0);
+  const updatedOptions = options ? { ...options, query } : { query };
+
+  const queryDataRef = useRef<QueryData<TData, TVariables>>();
+
+  function getQueryDataRef() {
+    if (!queryDataRef.current) {
+      queryDataRef.current = new QueryData<TData, TVariables>({
+        options: updatedOptions as QueryOptions<TData, TVariables>,
+        context,
+        forceUpdate
+      });
+    }
+    return queryDataRef.current;
+  }
+
+  const queryData = getQueryDataRef();
```

Comment:
The `getQueryDataRef` function doesn’t need to be a function if you’re only calling it here.

Reply:
Right! Left over from some previous refactoringmadness.